tokio/src/runtime/context/scoped.rs

```diff
@@ -0,0 +1,61 @@
+use std::cell::Cell;
+use std::marker::PhantomData;
+use std::ptr;
+
+/// Scoped thread-local storage
+pub(super) struct Scoped<T> {
+    pub(super) inner: Cell<*const ()>,
+    pub(crate) _p: PhantomData<T>,
```

Comment:
wait, why do we have a `Cell<*const ()>` _and_ a `PhantomData<T>`, and cast the pointer to `*const T` when it's actually dereferenced? can't this just be `Cell<*const T>`? Is this some kind of variance thing? If so, it would be nice to have a comment explaining what's up with this...

Reply:
Not sure; this was lifted from `scoped_tls`. I could see what happens.

Reply:
seems like it would be nicer that way, but 🤷‍♀️

Reply:
I switched it.